plugin/kubernetes/handler_test.go

```diff
@@ -408,6 +408,14 @@ var dnsTestCases = []kubeTestCase{
 			test.SOA("cluster.local.	5	IN	SOA	ns.dns.cluster.local. hostmaster.cluster.local. 1499347823 7200 1800 86400 5"),
 		},
 	}},
+	// A query for a double subdomain of an external service should not resolve to the external service
+	{Case: test.Case{
+		Qname: "double.endpoint.external.testns.svc.cluster.local.", Qtype: dns.TypeCNAME,
```

Comment:
OK, the fix is OK as is.  Though I think the use of the terms `double` and `endpoint` is confusing here.  They are really just subdomains.  In fact, if the `endpoint` segment was actually being parsed as an "endpoint", the existing check would have caught it.

```suggestion
		Qname: "subdomain1.subdomain2.external.testns.svc.cluster.local.", Qtype: dns.TypeCNAME,
```

Reply:
sounds good. I'll make the updates to the names so it's less confusing.